helm/flux-app/templates/servicemonitor.yaml

```diff
@@ -0,0 +1,39 @@
+{{- if .Values.serviceMonitor.enabled }}
+apiVersion: v1
+kind: Service
+metadata:
+  labels:
+    {{- include "labels.common" . | nindent 4 }}
+  name: {{ .Release.Name  }}-monitoring
+  namespace: {{ .Release.Namespace }}
+spec:
+  clusterIP: None
+  ports:
+    - name: http
+      port: 8080
+      protocol: TCP
+      targetPort: {{ .Values.upstreamMetricsPort }}
+  selector:
+    app.kubernetes.io/instance: {{ .Release.Name }}
+    app.kubernetes.io/managed-by: {{ .Release.Service }}
+    app.kubernetes.io/name: {{ include "name" . }}
+---
+apiVersion: monitoring.coreos.com/v1
+kind: ServiceMonitor
```

Comment:
Flux is one of the first things bootstrapped onto an empty cluster. What happens when `servicemonitors.monitoring.coreos.com` is not available, since it's enabled by default?

Reply:
Well it's part of what I wanted to test :)

Reply:
But I guess it will fail. I'm not sure how to get around that